controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/CommitRequestHandler.java

```diff
@@ -347,13 +355,21 @@ public CommitRequestHandler(final StreamMetadataStore streamMetadataStore,
                     // Note: if its a rerun, transaction commit offsets may have been updated already in previous iteration
                     // so this will not update/modify it. 
                     .thenCompose(v -> streamMetadataTasks.notifyTxnCommit(scope, stream, segments, txnId, context.getRequestId()))
-                    .thenCompose(map -> streamMetadataStore.recordCommitOffsets(scope, stream, txnId, map, context, executor))
+                    .thenCompose(map -> {
+                        Timer recordOffsetsTimer = new Timer();
+                        return streamMetadataStore.recordCommitOffsets(scope, stream, txnId, map, context, executor)
+                                .thenApply(x -> {
```

Comment:
`thenRun` - I don't see you're actually using `x` downstream

Reply:
fixed